doc/source/ray-core/scheduling/resources.rst

```diff
@@ -0,0 +1,153 @@
+  not the limiting factor for parallelism.
+
+Specifying Node Resources
+-------------------------
+
+By default, Ray nodes start with pre-defiend resources and the quantities of those resources on each node are set to the physical quantities auto detected by Ray.
+For example, if you start a head node with ``ray start --head`` then the quantity of logical CPU resources will be equal to the number of physical CPUs on the machine.
+However you can always override that by manually specifying the quantities of pre-defined resources and also add custom resources.
+There are several ways to do that depending on how you start the Ray cluster:
+
+.. tabbed:: ray.init()
+
+    If you are using :ref:`ray.init() <ray-init-ref>` to start a single node Ray cluster, you can do the following to manually specify node resources:
+
+    .. literalinclude:: ../doc_code/resources.py
+        :language: python
+        :start-after: __specifying_node_resources_start__
+        :end-before: __specifying_node_resources_end__
+
+.. tabbed:: ray start
+
+    If you are using :ref:`ray start <ray-start-doc>` to start a Ray node, you can run:
+
+    .. code-block:: shell
+
+        ray start --head --num-cpus=3 --num-gpus=4 --resources='{"special_hardware": 1, "custom_label": 1}'
+
+.. tabbed:: ray up
+
+    If you are using :ref:`ray up <ray-up-doc>` to start a Ray cluster, you can set the :ref:`resources field <cluster-configuration-resources-type>` in the yaml file:
+
+    .. code-block:: yaml
+
+        available_node_types:
+          head:
+            ...
+            resources:
+              CPU: 3
+              GPU: 4
+              special_hardware: 1
+              custom_label: 1
+
+.. tabbed:: kuberay
+
+    If you are using :ref:`kuberay <kuberay-index>` to start a Ray cluster, you can set the :ref:`rayStartParams field <rayStartParams>` in the yaml file:
+
+    .. code-block:: yaml
+
+        headGroupSpec:
+          rayStartParams:
+            num-cpus: "3"
+            num-gpus: "4"
+            resources: '"{\"special_hardware\": 1, \"custom_label\": 1}"'
+
+
+.. _resource-requirements:
+
+Specifying Task or Actor Resource Requirements
+----------------------------------------------
+
+Ray allows specifying a task or actor's resource requirements (e.g., CPU, GPU, and custom resources).
+The task or actor will only run on a node if there are enough required resources
+available to execute the task or actor.
+
+By default, Ray tasks use 1 CPU resource and Ray actors use 1 CPU for scheduling and 0 CPU for running
+(This means, by default, actors cannot get scheduled on a zero-cpu node, but an infinite number of them can run on any non-zero cpu node.
+If resources are specified explicitly, they are required
+for both scheduling and running. It's a confusing default behavior for actors due to historical reasons and it's recommended to always explicitly set ``num_cpus`` for actors to aovid any surprises).
```

Comment:
The sentence is accurate but for better public relations, maybe we should drop "confusing" and "historical reasons". Instead we can explain the historical reason. IIUC it's that we want to make sure that by default 0-CPU nodes don't schedule actors?

Reply:
The reason is that we want new people to be able to run many actors in the Jupiter notebook. When people try Ray, the notebook has very few CPUs and they may create many actors without destroying the old ones.

Reply:
I'm not sure we want to explain that details here.

Reply:
Yeah, ok, confusing and historical is appropriate. Seems there's a fundamental issue that wasn't quite thought out.